### PR TITLE
fix: show default name of common date filter even for empty string

### PR DESCRIPTION
**Pull request by xMort**
JIRA: LX-233

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18888/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20610/ : SUCCESS in 14m 12s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFilterConfigurationState.ts [hunk 1]
```diff
@@ -48,7 +47,7 @@ export function useDependentDateFilterConfigurationState(
         const commonDate = parseCommonDateFilter(commonDateFilter);
 
         return (filterElementsByDate ?? [])
-            .filter((parent) => !!parent.isCommonDate)
+            .filter((parent) => parent.isCommonDate)
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useDependentDateFilterConfigurationState.ts:50]: These were underlined by eslint in IDE as previously the type was `boolean | undefined` but was later changed to `boolean` and boolean re-casting operators remained in place even when they are no longer necessary.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts [hunk 2]
```diff
@@ -263,7 +263,7 @@ export const useFilterItems = (
 
             const commonDateFilter: IValuesLimitingItemWithTitle[] = [
                 {
-                    title: filterConfig?.filterName ?? intl.formatMessage({ id: "dateFilterDropdown.title" }),
+                    title: filterConfig?.filterName || intl.formatMessage({ id: "dateFilterDropdown.title" }),
```
Comment (xMort) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts:266]: Actual fix: the filter name was an empty string in the workspace where the issue manifested.

**Review by pbenes — APPROVED** [review]

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/2139/ : SUCCESS in 1m 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/20611/ : SUCCESS in 7m 25s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1938/ : SUCCESS in 13m 45s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1918/ : SUCCESS in 6m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5436/ : SUCCESS in 6s

**Pull request merged**